### ISSUE TITLE
JFrog Tools Installer Task does not install the latest CLI version

### ISSUE THREAD

**Issue by vijayarohinipendyala** [jfrog/artifactory-vsts-extension#348]
Issue1: Currently in our Azure DevOps Server 2019, we have installed Jfrog.jfrog-azure-devops-extension-2.1.1.vsix extension.

In the JFrog Tools Installer task, we updated the Artifactory Service Connection and also passed the Artifactory remote repository which mimics the jfrog-cli/v2-jf repository.

This task only installs 2.15.0 cli version and not the 2.18.2 version.

If we enable the custom version and pass the version number, then it downloads that version.  
What is the default CLI version that it should download if we do not pass the custom version?

Note:  Our current Artifactory version is 6.x

Issue 2: In our on-prem Artifactory, we have a Virtual repository which has two  remote repositories that mimic:  
https://releases.jfrog.io/artifactory/jfrog-cli/v1/
https://releases.jfrog.io/artifactory/jfrog-cli/v2/

We have this extension: JFrog Artifactory Extension 1.13.2 installed and the associated Tools Installer task only installs JFrog CLI version 1.52.0 and not 1.53.2 version or 2.18.2 (JFrog.exe) from the remote repository. 

Tried deleting _jf, jf, _jfrog and jfrog directories from the Azure Devops Build Server, still same behavior.

Would the Tools Installer Task associated to 1.13.2 version pull in the JFrog.exe file from the remote repository jfrog-cli/v2/ ?

Thank you,

**Comment by RobiNino** [comment]
Hi @vijayarohinipendyala ,
We recommend using the newer extension, JFrog Azure DevOps Extension, to enjoy all the new features and capabilities.

| Extension | Supported JFrog CLI version | JFrog CLI remote download URL |
|:------:|:-----:|:-----:|
| JFrog Azure DevOps | JFrog CLI v2 | https://releases.jfrog.io/artifactory/jfrog-cli/v2-jf/ |
| Artifactory Azure DevOps | JFrog CLI v1 | https://releases.jfrog.io/artifactory/jfrog-cli/v1/ |

(Note that the URL for v2 is slightly different than the one you mentioned).

The JFrog CLI version downloaded isn't the latest by design. The extension [keeps a default version](https://github.com/jfrog/jfrog-azure-devops-extension/blob/e414c87d3c512a2aa522278483526a0fbcfc1d8a/jfrog-tasks-utils/utils.js#L12) that is updated on every extension release. You may download a different version using the custom  version checkbox as you mentioned.

Let me know if you have any other questions regarding this.
Thanks

**Comment by vijayarohinipendyala** [comment]
hi @RobiNino,

Thank you for replying back!
Have few more questions:
1) Do we have the timeline on when the JFrog CLI V1 will be sunset?
2)  We are currently using JFrog Artifactory extension version 1.13.2,   where we initially use the "Artifactory Tools installer" task to download the CLI and then use the "Artifact Generic Download" or "Artifact Generic Upload" or Artifact Build Promotion tasks etc in our ADS pipelines.

Lately observed a sporadic issue, where the "Artifact Generic Download" or "Artifact Generic Upload" or Artifact Build Promotion tasks fails with 401 Unauthorized error, but when we redeploy the same pipeline without making any changes works fine.
Do you think this might be related to the JFrog CLI version?

Thank you,
Ro

**Comment by eyalbe4** [comment]
@vijayarohinipendyala,
There's no timeline for the sunset for JFrog CLI v1 for now, but moving to v2 is definitely the way to go, due multiple improvements and mew features. 
As for the issue you mentioned, we haven't seen it or received reports for such an issue. It probably requires a closer look, checking why the 401 is returned initially.

**Comment by fourpastmidnight** [comment]
Yes, but _even the newer extension_ *DOES NOT* ship the latest JFrog CLI v2 with each release! I found this out the hard way when a bug in the CLI was "fixed", and I updated the extension, but the bug was still present...only to discover the _OLD_ *broken* CLI version was being used. _Extremely frustrating and a complete waste of time!_
